rudder-web/src/main/scala/com/normation/rudder/web/components/DirectiveEditForm.scala

```diff
@@ -178,7 +178,15 @@ class DirectiveEditForm(
   val displayDeprecationWarning = technique.deprecrationInfo match {
     case Some(info) =>
       ( "#deprecation-message *" #> info.message &
-        "#migrate-button *" #> migrateButton(fullActiveTechnique.techniques.keys.max,"Migrate now!","deprecation-migration")
+        "#migrate-button *" #> {
+            val lastTechniqueVersion = fullActiveTechnique.techniques.keys.max
```

Comment:
Is it possible that a technique may have no technique version ? 

Here a directive has already been selected meaning that we have already built a directive based on a existing version of the technique.

I guess it's impossible in a normal usage of Rudder, but i think it's still possible by manipulating ldap to have no version and still one in the technique